File: examples/alexa-hello-world-java/src/main/java/com/amazon/ask/helloworld/handlers/HelloWorldIntentHandler.java
package com.amazon.ask.helloworld.handlers;

import com.amazon.ask.dispatcher.request.handler.HandlerInput;
import com.amazon.ask.dispatcher.request.handler.RequestHandler;
import com.amazon.ask.model.Response;
import com.amazon.ask.request.Predicates;

import javax.inject.Singleton;
import java.util.Optional;

@Singleton
public class HelloWorldIntentHandler implements RequestHandler {

    private final MessageService messageService;

    public HelloWorldIntentHandler(MessageService messageService) {
        this.messageService = messageService;
    }

    @Override
    public boolean canHandle(HandlerInput input) {
        return input.matches(Predicates.intentName("HelloWorldIntent"));
    }

    @Override
    public Optional<Response> handle(HandlerInput input) {
        String speechText = messageService.sayHello();
        return input.getResponseBuilder()
                .withSpeech(speechText)
                .withSimpleCard("HelloWorld", speechText)
                .build();
    }

}

File: test-suite-kotlin/src/test/kotlin/io/micronaut/docs/aws/alexa/handlers/HelloWorldIntentHandler.kt
package io.micronaut.docs.aws.alexa.handlers

//tag::clazz[]
import com.amazon.ask.dispatcher.request.handler.HandlerInput
import com.amazon.ask.dispatcher.request.handler.RequestHandler
import com.amazon.ask.model.Response
import com.amazon.ask.request.Predicates
import jakarta.inject.Singleton
import java.util.Optional

@Singleton // <1>
class HelloWorldIntentHandler : RequestHandler {

    override fun canHandle(input: HandlerInput): Boolean {
        return input.matches(Predicates.intentName("HelloWorldIntent"))
    }

    override fun handle(input: HandlerInput): Optional<Response> {
        val speechText = "Hello world"
        return input.responseBuilder
            .withSpeech(speechText)
            .withSimpleCard("HelloWorld", speechText)
            .build()
    }

}
//end::clazz[]
